app/uk/gov/hmrc/gform/validation/ComponentValidator.scala

```diff
@@ -400,4 +405,37 @@ object ComponentValidator {
     if (maybeCode === expectedCode.map(_.code)) validationSuccess else emailError
 
   }
+
+  def validateTime(fieldValue: FormComponent, time: Time)(data: FormDataRecalculated)(
+    implicit messages: Messages,
+    l: LangADT,
+    sse: SmartStringEvaluator): ValidatedType[Unit] = {
+    val timeValue = data.data.get(fieldValue.id).toSeq.flatMap(_.toSeq).filterNot(_.isEmpty).headOption
+
+    val twelveHoursFormat = DateTimeFormatter.ofPattern("hh:mm a")
+
+    @tailrec
+    def getTimeSlots(sTime: LocalTime, eTime: LocalTime, iMins: Int, acc: List[LocalTime]): List[LocalTime] = {
+      val t = sTime.plusMinutes(iMins)
+      if (t.isAfter(eTime) || (0 until iMins contains MINUTES
+            .between(LocalTime.parse("00:00"), t)))
+        acc
+      else
+        getTimeSlots(t, eTime, iMins, acc :+ t)
+    }
+
+    lazy val timeSlots = time.ranges
+      .flatMap(t =>
+        getTimeSlots(t.startTime.time, t.endTime.time, time.intervalMins.intervalMins, List(t.startTime.time)))
+      .distinct
+      .map(_.format(twelveHoursFormat))
```

Comment:
This is repeated twice in this PR.

Why not to do it an instance method of `case class Range`?

(No need to have it on backend)